web/src/components/molecules/Asset/AssetList/index.tsx

```diff
@@ -56,18 +70,32 @@ const AssetList: React.FC<Props> = ({
     setUploadModalVisibility(true);
   }, [setUploadModalVisibility]);
 
-  const hideUploadModal = useCallback(() => {
-    setUploadModalVisibility(false);
-    setUploading(false);
-    setFileList([]);
-  }, [setFileList, setUploading, setUploadModalVisibility]);
-
-  const handleUpload = useCallback(() => {
+  const handleUpload = useCallback(async () => {
     setUploading(true);
-    onAssetCreate(fileList).finally(() => {
+
```

Comment:
On line 74, you setUploading to true, but then it seems like you don't set it to false when finished

Reply:
Also, is uploading/setUploading only used here? We have a lot of props being passed as it stands, and it would be nice to keep state exactly where it is needed.
If it is only needed here, in AssetList/index.tsx, consider moving the declaration here (and any other state/props that aren't needed so high up, and then creating a local handleOnModalClose or whatever callback is needed to setUploading(false)

Reply:
setUploading to false is done inside `hideUploadModal` function.

`uploading` and `setUploading` are used in multiple places.

Reply:
Anyways, I've just made a refactor to use setUploading only from one place.